expression/builtin_encryption_test.go

```diff
@@ -398,17 +398,26 @@ func decodeHex(str string) []byte {
 func TestCompress(t *testing.T) {
 	t.Parallel()
 	ctx := createContext(t)
+	fc := funcs[ast.Compress]
+	gbkStr, _ := charset.NewEncoding("gbk").EncodeString("你好")
 	tests := []struct {
+		chs    string
 		in     interface{}
 		expect interface{}
 	}{
-		{"hello world", string(decodeHex("0B000000789CCA48CDC9C95728CF2FCA4901040000FFFF1A0B045D"))},
-		{"", ""},
-		{nil, nil},
+		{"", "hello world", string(decodeHex("0B000000789CCA48CDC9C95728CF2FCA4901040000FFFF1A0B045D"))},
+		{"", "", ""},
+		{"", nil, nil},
+		{"utf8mb4", "hello world", string(decodeHex("0B000000789CCA48CDC9C95728CF2FCA4901040000FFFF1A0B045D"))},
+		{"gbk", "hello world", string(decodeHex("0B000000789CCA48CDC9C95728CF2FCA4901040000FFFF1A0B045D"))},
+		{"utf8mb4", "你好", string(decodeHex("06000000789C7AB277C1D3A57B01010000FFFF10450489"))},
+		{"gbk", gbkStr, string(decodeHex("04000000789C3AF278D76140000000FFFF07F40325"))},
 	}
-
-	fc := funcs[ast.Compress]
 	for _, test := range tests {
+		if len(test.chs) != 0 {
+			err := ctx.GetSessionVars().SetSystemVar(variable.CharacterSetConnection, test.chs)
```

Comment:
this didn't take any effect, or change the datum collation

Reply:
Done